lib/modules/platform/gerrit/index.ts

```diff
@@ -0,0 +1,543 @@
+  EnsureIssueConfig,
+  EnsureIssueResult,
+  FindPRConfig,
+  Issue,
+  MergePRConfig,
+  PlatformParams,
+  PlatformResult,
+  Pr,
+  RepoParams,
+  RepoResult,
+  UpdatePrConfig,
+} from '../types';
+import { repoFingerprint } from '../util';
+
+import { smartTruncate } from '../utils/pr-body';
+import { readOnlyIssueBody } from '../utils/read-only-issue-body';
+import { client } from './client';
+import { configureScm } from './scm';
+import type {
+  GerritChange,
+  GerritFindPRConfig,
+  GerritLabelTypeInfo,
+  GerritProjectInfo,
+} from './types';
+import {
+  TAG_PULL_REQUEST_BODY,
+  getGerritRepoUrl,
+  mapBranchStatusToLabel,
+  mapGerritChangeToPr,
+  mapPrStateToGerritFilter,
+} from './utils';
+
+export const id = 'gerrit';
+
+const defaults: {
+  endpoint?: string;
+} = {};
+
+let config: {
+  repository?: string;
+  head?: string;
+  config?: GerritProjectInfo;
+  labels: Record<string, GerritLabelTypeInfo>;
+  labelMappings?: {
+    stabilityDaysLabel?: string;
+    mergeConfidenceLabel?: string;
+  };
+  gerritUsername?: string;
+} = {
+  labels: {},
+  labelMappings: {
+    stabilityDaysLabel: 'Renovate-Stability',
+  },
+};
+
+export function writeToConfig(newConfig: typeof config): void {
+  config = { ...config, ...newConfig };
+}
+
+export function initPlatform({
+  endpoint,
+  username,
+  password,
+  gerritLabelMapping,
+}: PlatformParams & RepoGlobalConfig): Promise<PlatformResult> {
+  logger.debug(`initPlatform(${endpoint!}, ${username!})`);
+  if (!endpoint) {
+    throw new Error('Init: You must configure a Gerrit Server endpoint');
+  }
+  if (!(username && password)) {
+    throw new Error(
+      'Init: You must configure a Gerrit Server username/password'
+    );
+  }
+  config.labelMappings = gerritLabelMapping;
+  config.gerritUsername = username;
+  defaults.endpoint = ensureTrailingSlash(endpoint);
+  setBaseUrl(defaults.endpoint);
+  const platformConfig: PlatformResult = {
+    endpoint: defaults.endpoint,
+  };
+  return Promise.resolve(platformConfig);
+}
+
+/**
+ * Get all state="ACTIVE" and type="CODE" repositories from gerrit
+ */
+export async function getRepos(): Promise<string[]> {
+  logger.debug(`getRepos()`);
+  return await client.getRepos();
+}
+
+/**
+ * Clone repository to local directory
+ * @param config
+ */
+export async function initRepo({
+  repository,
+  gitUrl,
+}: RepoParams): Promise<RepoResult> {
+  logger.debug(`initRepo(${repository}, ${gitUrl!})`);
+  const projectInfo = await client.getProjectInfo(repository);
+  const branchInfo = await client.getBranchInfo(repository);
+
+  config = {
+    ...config,
+    repository,
+    head: branchInfo.revision,
+    config: projectInfo,
+    labels: projectInfo.labels ?? {},
+  };
+  const baseUrl = defaults.endpoint!;
+  const url = getGerritRepoUrl(repository, baseUrl);
+  configureScm(repository, config.gerritUsername!);
+  await git.initRepo({ url });
+
+  //abandon "open" and "rejected" changes at startup
```

Comment:
why do we do this? wouldn't this cause all updates to be immortal? So renovate always recreates the update.

Reply:
No, the `prAlreadyExisted` function still find these closed changes and skip re-creation.
It should work with Gerrit like described [here](https://docs.renovatebot.com/key-concepts/pull-requests/#how-renovate-finds-existing-prs).